make/addon.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# Helper script for addons: deploy and destroy of argocd, fluxcd, tsb-monitoring and external-dns.
+
+export BASE_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source ${BASE_DIR}/helpers.sh
+
+ACTION=${1}
+CLOUD_PROVIDER=${2}
+ADDON_NAME=${3}
+
+# This function provides help information for the script.
+function help() {
+  echo "Usage: $0 <command> [options]"
+  echo "Commands:"
+  echo "  help                                        Display this help message."
+  echo "  deploy_addon  <cloud_provider> <addon_name> Deploy addon on the specified cloud provider."
+  echo "  destroy_addon <cloud_provider> <addon_name> Destroy addon on the specified cloud provider."
+}
+
+# Check if the input parameters are correctly given.
+if [[ -z "${ACTION}" ]]; then
+  print_error "No command provided."
+  help
+  exit 1
+fi
+
+# Check if TFVARS_JSON is not defined and exit.
+if [ -z "${TFVARS_JSON}" ]; then
+  print_error "TFVARS_JSON is not defined."
+  exit 1
+fi
+
+# Check if the file pointed to by TFVARS_JSON does not exist and exit.
+if [ ! -f "${TFVARS_JSON}" ]; then
+  print_error "File '${TFVARS_JSON}' does not exist."
+  exit 1
+fi
+
+# Validate input values.
+if [[ "${ACTION}" == "deploy_addon" || "${ACTION}" == "destroy_addon" ]]; then
```

Comment:
let's avoid complex IFs for the future maintainability in favour of arrays like

```
CLOUD_PROVIDER="aws"
CLOUD_PROVIDERS=("aws" "azure" "gcp")

# Check if "a" is in the array
if [[ ! " ${CLOUD_PROVIDERS[*]} " == *" $CLOUD_PROVIDER "* ]]; then
    print_error "Invalid cloud provider. Must be one of 'azure', 'aws', or 'gcp'."
fi
```

Reply:
Ack, adopted